xdsl/printer.py

```diff
@@ -652,6 +652,26 @@ def print_op_attributes(self, attributes: dict[str, Attribute]) -> None:
 
         self.print("}")
 
+    def print_op_attributes_with_keyword(
```

Comment:
It feels like this could be fused with the previous method, with a pretty low overhead. I'd just check whether reserved attr names is not empty before filtering the attribute list

Reply:
Oh and please remove the `if len(attributes) == 0` and replace with `if not attributes`